diff --git a/session-loader/Development/IDE/Session.hs b/session-loader/Development/IDE/Session.hs
index bdd27f3d..2639631d 100644
--- a/session-loader/Development/IDE/Session.hs
+++ b/session-loader/Development/IDE/Session.hs
@@ -880,7 +880,7 @@ newComponentCache recorder exts cradlePath _cfp hsc_env old_cis new_cis = do
               getSession
 #endif
       henv <- createHscEnvEq thisEnv (zip uids dfs)
-      let targetEnv = (if isBad ci then multi_errs else [], Just henv)
+      let targetEnv = ([], Just henv)
           targetDepends = componentDependencyInfo ci
           res = ( targetEnv, targetDepends)
       logWith recorder Debug $ LogNewComponentCache res
